### ISSUE TITLE
Board / Library Manager] split ridiculus long menus into sub-menus

### ISSUE THREAD

**Issue by datenheim** [arduino/Arduino#8858]
The menu list of the board-manager is very (too?) long and scrolling in this menu is not convenient and seems to have (had) many bugs.

Indeed the Boards already are grouped by "Arduino AVR-Boards" etc.
These groups could/should simply be transformed to a submenu.

Thus boards are simpler and quicker to reach and scroll'n'search is reduced largely.

Same may happen with the library manager. There also a grouping is existing and could be used.
Libraries not fitting into the three groups could get an own submenu.

I have only ESP32, ESP8266 and Micro/Mini/MegaCore Boardmanagers and even on FHD+ screen the menu is 3x longer as the screen. Scrolling really takes some time and simply the wrong solution to this problem.

Related to all versions up to 1.8.9 on Windows and afaik also on Linux.

**Comment by per1234** [comment]
>Scrolling really takes some time and simply the wrong solution to this problem.

Although not what you're requesting, this issue will be somewhat alleviated via https://github.com/arduino/Arduino/pull/8607

**Comment by datenheim** [comment]
Yes, the feature mentioned in #8607 presenting recently used boards (libraries) more prominently will fit nicely with my request. There could well be a first submenu showing the recently used objects.

**Comment by matthijskooijman** [comment]
For libraries, I implemented something similar in #1986, which divides puts libraries into submenus based on the current heading distinctions, as well as allowing them to be organized into subfolders under "libraries" (and these subfolders also cause submenus). The pull request was recently closed though, and was already quite outdated, but might still serve as inspiration in the future.

**Comment by per1234** [comment]
The **Tools > Board** component of this request was fixed by https://github.com/arduino/Arduino/pull/9238 (thanks Matthijs!). The **Sketch > Include Library** component remains.
